app/controllers/homepage_controller.rb

```diff
@@ -0,0 +1,18 @@
+class HomepageController < ApplicationController
+  skip_before_action :authorize
+
+  before_action :discard_current_user_questions, only: [:index], if: :signed_in?
+
+  def index
+    @q = Question.includes(:user, :rich_text_content).published_questions.chronological_order.ransack(params[:q])
+    @questions = @q.result
+  end
+
+  private def discard_current_user_questions
+    if params[:q].present?
```

Comment:
instead of if else can we simply use,
params[:q][:user_id_not_eq] ||= { user_id_not_eq: current_user.id }

Reply:
URL like 127.0.0.1:3000 wont have q param so param[:q] will be undefined as well hence i created params[:q] check